src/utils/shallowEqual.js

```diff
@@ -23,28 +23,26 @@ function shallowEqual(objA, objB) {
     return true;
   }
 
-  if (!objA || !objB) {
+  if (typeof objA !== 'object' || objA === null ||
+      typeof objB !== 'object' || objB === null) {
     return false;
   }
 
-  if (typeof objA !== 'object' || typeof objB !== 'object') {
+  var keysA = Object.keys(objA);
+  var keysB = Object.keys(objB);
+
+  if (keysA.length !== keysB.length) {
     return false;
   }
 
-  var key;
   // Test for A's keys different from B.
-  for (key in objA) {
-    if (objA.hasOwnProperty(key) &&
-        (!objB.hasOwnProperty(key) || objA[key] !== objB[key])) {
-      return false;
-    }
-  }
-  // Test for B's keys missing from A.
-  for (key in objB) {
-    if (objB.hasOwnProperty(key) && !objA.hasOwnProperty(key)) {
+  var bHasOwnProperty = Object.prototype.hasOwnProperty.bind(objB);
```

Comment:
Hmm, I'm not sure it'll matter *too* much. When I think about this, creating 1 bound function to reuse seems cheaper than making N calls using `call` which I assume ends up creating a new execution context on each call.